File: app/src/main/java/com/zeapo/pwdstore/di/components/AppComponent.java
package com.zeapo.pwdstore.di.components;

import android.app.Application;

import com.zeapo.pwdstore.PasswordStoreApplication;
import com.zeapo.pwdstore.di.builder.ActivityBuilder;
import com.zeapo.pwdstore.di.modules.FragmentModule;
import com.zeapo.pwdstore.di.modules.RandomModule;
import com.zeapo.pwdstore.di.modules.RandomPasswordGeneratorModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

@Singleton
@Component(modules = {AndroidInjectionModule.class, ActivityBuilder.class})
public interface AppComponent {
    void inject(PasswordStoreApplication app);

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }
}

File: app/src/main/java/com/zeapo/pwdstore/di/AppComponent.kt
/*
 * Copyright © 2014-2020 The Android Password Store Authors. All Rights Reserved.
 * SPDX-License-Identifier: GPL-3.0-only
 */
package com.zeapo.pwdstore.di

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager
import com.zeapo.pwdstore.PasswordStore
import dagger.BindsInstance
import dagger.Component
import dagger.Module
import dagger.Provides
import dagger.Reusable
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {

    @Component.Factory
    interface Factory {
        fun create(@BindsInstance applicationContext: Context): AppComponent
    }

    fun inject(activity: PasswordStore)
}

@Module
object AppModule {

    @Provides
    @Reusable
    fun provideSharedPrefs(context: Context): SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
}
